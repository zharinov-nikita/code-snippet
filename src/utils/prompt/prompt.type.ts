import { PromptObject } from 'prompts'

export type TypeQuestion = PromptObject[]
