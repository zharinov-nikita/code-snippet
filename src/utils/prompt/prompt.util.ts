import { prompt } from 'prompts'
import type { TypeArgs } from '../../types'
import { UtilConfig } from '../config'
import type { TypeQuestion } from './prompt.type'

export class UtilPrompt {
  private config: UtilConfig

  constructor() {
    this.config = new UtilConfig()
  }

  private createQuestion(): TypeQuestion {
    return [
      {
        type: 'select',
        name: 'snippet',
        message: 'Pick a snippet',
        choices: this.config.generate().map(({ snippet }) => ({ title: snippet, value: snippet })),
        initial: 0,
      },
      { type: 'text', name: 'name', message: 'Pick a name (required)' },
      { type: 'text', name: 'path', message: 'Pick a path (required)' },
      {
        type: 'toggle',
        name: 'isFlat',
        message: 'Create a flat file structure?',
        initial: false,
        active: 'yes',
        inactive: 'no',
      },
      { type: 'text', name: 'prefix', message: 'Pick a prefix (optional - to skip, press enter)' },
      { type: 'text', name: 'suffix', message: 'Pick a suffix (optional - to skip, press enter)' },
    ]
  }

  public async getArgs(): Promise<TypeArgs> {
    const question = this.createQuestion()
    const args = await prompt(question)
    return args
  }
}
