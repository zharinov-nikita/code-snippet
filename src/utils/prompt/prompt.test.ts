import { describe, expect, test } from '@jest/globals'
import { UtilPrompt } from './prompt.util'

describe('util/prompt', () => {
  const util = new UtilPrompt()

  test('snapshot', () => {
    expect(util).toMatchSnapshot('util')
  })
})
