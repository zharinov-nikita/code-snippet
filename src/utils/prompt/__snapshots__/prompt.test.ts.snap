// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`util/prompt snapshot: util 1`] = `
UtilPrompt {
  "config": UtilConfig {},
}
`;
